src/components/AccountForms/LoginForm/index.js

```diff
@@ -0,0 +1,192 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { API_PATH } from 'utils/constants';
+import request from 'utils/axiosConfig';
+import { useAuth } from '../../../context/auth';
+import { GlobalContext } from 'context/global';
+import isEmpty from 'lodash/isEmpty';
+
+import { Link, useHistory, useLocation } from 'react-router-dom';
+import useStyles from './LoginForm.styles';
+import {
+  Paper,
+  TextField,
+  InputAdornment,
+  IconButton,
+  Button,
+} from '@material-ui/core';
+import { Visibility, VisibilityOff, Error } from '@material-ui/icons';
+import { ReactComponent as LoginHero } from './assets/LoginHero.svg';
+import { ReactComponent as TwitterLogo } from './assets/TwitterLogo.svg';
+
+const FORM_TITLE = 'Welcome back!';
+
+const emailRequirements = {
+  required: 'Email is required',
+  pattern: {
+    value: /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+    message: 'Invalid email format',
+  },
+};
+
+const LoginForm = ({ testOnSubmit }) => {
+  const [showPassword, setShowPassword] = useState(false);
+  const { register, handleSubmit, errors } = useForm();
+  const {
+    globalState: { isMobile },
+  } = useContext(GlobalContext);
+  const history = useHistory();
+  const { search } = useLocation();
+  const classes = useStyles({
+    isMobile: isMobile,
+  });
+
+  const { auth } = useAuth();
+
+  useEffect(() => {
+    if (auth.isLoggedIn()) {
+      history.replace('/mentors');
+    }
+  }, [auth, history]);
+
+  const onSubmit = (credentials) => {
+    request.post(API_PATH.LOGIN, credentials).then(({ status }) => {
+      if (status === 200) {
+        auth.checkCookie();
+        history.push('/mentors');
+      }
+    });
+  };
+
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const handleClickRegister = () => {
+    history.push('/register');
+  };
+
+  const content = (
+    <div className={classes.root}>
+      <Paper classes={{ root: classes.paperContainer }}>
+        <LoginHero className={classes.heroImage} />
+        <div className={classes.loginFormContainer}>
+          {!isEmpty(errors) && !isMobile && (
+            <div className={`${classes.alert} ${classes.error}`}>
+              <Error />
+              <p role="alert">Sorry, invalid email or password. Try again?</p>
+            </div>
+          )}
+          {isEmpty(errors) && search === '?timedOut=true' && (
```

Comment:
### Change requested:
- used the extracted value of the query string here

`{isEmpty(errors) && timedOut && (`     ....